### ISSUE TITLE
79360.patch has wrong hashsum

### ISSUE THREAD

**Issue by petronny** [rocm-arch/python-pytorch-rocm#48]
```
    fix-building-for-torchvision.patch ... Passed
    79360.patch ... FAILED
    ffmpeg4.4.patch ... Passed
    rocblas.patch ... Skipped
==> ERROR: One or more files did not pass the validity check!
==> ERROR: Could not download sources.
```
https://github.com/arch4edu/cactus/runs/7686453950?check_suite_focus=true

**Comment by acxz** [comment]
Resolved with [997e653](https://github.com/rocm-arch/python-pytorch-rocm/commit/997e6535c0334f2286221e99a16b27cb85366382)

**Comment by IshiRenga** [comment]
I'm also seeing the 79360.patch fail even with the updated hashsum

**Comment by acxz** [comment]
Is the patch failing to checksum or is it failing to patch the files?

**Comment by IshiRenga** [comment]
Oh sorry, I should have specified that it is failing the checksum

```
    use-system-libuv.patch ... Passed
    fix-building-for-torchvision.patch ... Passed
    79360.patch ... FAILED
    ffmpeg4.4.patch ... Passed
    rocblas.patch ... Skipped
==> ERROR: One or more files did not pass the validity check!
```

**Comment by acxz** [comment]
Thx, I think the checksum probably got updated. I'll just skip the checksums check for the PR.

**Comment by IshiRenga** [comment]
Thank you for the quick response time. The updated version worked and allowed me to start compiling `python-pytorch-rocm`, but unfortunately I ran into errors while compiling it. [Here](https://pastebin.com/aas9WaLT) is a pastebin of the build log near the end when I ran `paru -S python-pytorch-rocm python-torchvision-rocm`. Note I have set `export PYTORCH_ROCM_ARCH=gfx1010` before starting the compile. It says it failed to compile `benchmark.cc.o` and `FbgemmSparseDenseInt8Avx2.cc.o`. Here are just the stacktraces:

```
[3857/6663] Building CXX object third_party/benchmark/src/CMakeFiles/benchmark.dir/benchmark.cc.o
FAILED: third_party/benchmark/src/CMakeFiles/benchmark.dir/benchmark.cc.o 
/usr/bin/c++ -DHAVE_POSIX_REGEX -DHAVE_STD_REGEX -DHAVE_STEADY_CLOCK -I/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/benchmark/include -I/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/benchmark/src -I/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/benchmark/src/../include -isystem /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/cmake/../third_party/googletest/googlemock/include -isystem /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/cmake/../third_party/googletest/googletest/include -isystem /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/protobuf/src -isystem /opt/intel/mkl/include -isystem /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/gemmlowp -isystem /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/neon2sse -isystem /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/XNNPACK/include -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -fvisibility-inlines-hidden -DUSE_PTHREADPOOL  -std=c++11  -Wall  -Wextra  -Wshadow  -Wsuggest-override  -pedantic  -pedantic-errors  -fstrict-aliasing  -Wno-deprecated-declarations  -Wstrict-aliasing -O3 -DNDEBUG  -Werror  -Wno-deprecated   -march=x86-64 -std=gnu++14 -MD -MT third_party/benchmark/src/CMakeFiles/benchmark.dir/benchmark.cc.o -MF third_party/benchmark/src/CMakeFiles/benchmark.dir/benchmark.cc.o.d -o third_party/benchmark/src/CMakeFiles/benchmark.dir/benchmark.cc.o -c /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/benchmark/src/benchmark.cc
In file included from /usr/include/c++/12.1.1/string:40,
                 from /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/benchmark/include/benchmark/benchmark.h:185,
                 from /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/benchmark/src/benchmark.cc:15:
In static member function ‘static std::char_traits<char>::char_type* std::char_traits<char>::copy(char_type*, const char_type*, std::size_t)’,
    inlined from ‘static void std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_S_copy(_CharT*, const _CharT*, size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at /usr/include/c++/12.1.1/bits/basic_string.h:422:21,
    inlined from ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_M_replace(size_type, size_type, const _CharT*, size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at /usr/include/c++/12.1.1/bits/basic_string.tcc:532:22,
    inlined from ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::assign(const _CharT*) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at /usr/include/c++/12.1.1/bits/basic_string.h:1646:19,
    inlined from ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator=(const _CharT*) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at /usr/include/c++/12.1.1/bits/basic_string.h:814:28,
    inlined from ‘size_t benchmark::RunSpecifiedBenchmarks(BenchmarkReporter*, BenchmarkReporter*)’ at /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/benchmark/src/benchmark.cc:448:12:
/usr/include/c++/12.1.1/bits/char_traits.h:431:56: error: ‘void* __builtin_memcpy(void*, const void*, long unsigned int)’ accessing 9223372036854775810 or more bytes at offsets [2, 9223372036854775807] and 1 may overlap up to 9223372036854775813 bytes at offset -3 [-Werror=restrict]
  431 |         return static_cast<char_type*>(__builtin_memcpy(__s1, __s2, __n));
      |                                        ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
[3870/6663] Building CXX object third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/EmbeddingSpMDM.cc.o
ninja: build stopped: subcommand failed.
```
```
[32/2794] Building CXX object third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/FbgemmSparseDenseInt8Avx2.cc.o
FAILED: third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/FbgemmSparseDenseInt8Avx2.cc.o 
/usr/bin/c++ -DFBGEMM_STATIC -I/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/cpuinfo/include -I/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/third_party/asmjit/src -I/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/include -I/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm -I/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/cmake/../third_party/benchmark/include -isystem /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/cmake/../third_party/googletest/googlemock/include -isystem /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/cmake/../third_party/googletest/googletest/include -isystem /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/protobuf/src -isystem /opt/intel/mkl/include -isystem /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/gemmlowp -isystem /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/neon2sse -isystem /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/XNNPACK/include -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -Wall -Wextra -Werror -Wno-deprecated-declarations -O3 -DNDEBUG -fPIC -fvisibility=hidden   -march=x86-64 -m64 -mavx2 -mf16c -mfma -std=c++14 -MD -MT third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/FbgemmSparseDenseInt8Avx2.cc.o -MF third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/FbgemmSparseDenseInt8Avx2.cc.o.d -o third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/FbgemmSparseDenseInt8Avx2.cc.o -c /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/12.1.1/include/immintrin.h:47,
                 from /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc:11:
In function ‘void _mm256_maskstore_epi32(int*, __m256i, __m256i)’,
    inlined from ‘void fbgemm::internal::SparseDenseInt8MMAvx2(int, const std::unique_ptr<fbgemm::BCSRMatrix<> >&, const uint8_t*, int, int32_t*, uint8_t*, int, fbgemm::trRequantizationParams_t&, bool, int, int) [with bool FUSE_RELU = true; fbgemm::QuantizationGranularity Q_GRAN = fbgemm::QuantizationGranularity::TENSOR]’ at /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc:190:35:
/usr/lib/gcc/x86_64-pc-linux-gnu/12.1.1/include/avx2intrin.h:1155:32: error: ‘mask_int32_v’ may be used uninitialized [-Werror=maybe-uninitialized]
 1155 |   __builtin_ia32_maskstored256 ((__v8si *)__X, (__v8si)__M, (__v8si)__Y);
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc: In function ‘void fbgemm::internal::SparseDenseInt8MMAvx2(int, const std::unique_ptr<fbgemm::BCSRMatrix<> >&, const uint8_t*, int, int32_t*, uint8_t*, int, fbgemm::trRequantizationParams_t&, bool, int, int) [with bool FUSE_RELU = true; fbgemm::QuantizationGranularity Q_GRAN = fbgemm::QuantizationGranularity::TENSOR]’:
/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc:165:19: note: ‘mask_int32_v’ was declared here
  165 |           __m256i mask_int32_v;
      |                   ^~~~~~~~~~~~
In function ‘void _mm256_maskstore_epi32(int*, __m256i, __m256i)’,
    inlined from ‘void fbgemm::internal::SparseDenseInt8MMAvx2(int, const std::unique_ptr<fbgemm::BCSRMatrix<> >&, const uint8_t*, int, int32_t*, uint8_t*, int, fbgemm::trRequantizationParams_t&, bool, int, int) [with bool FUSE_RELU = true; fbgemm::QuantizationGranularity Q_GRAN = fbgemm::QuantizationGranularity::OUT_CHANNEL]’ at /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc:190:35:
/usr/lib/gcc/x86_64-pc-linux-gnu/12.1.1/include/avx2intrin.h:1155:32: error: ‘mask_int32_v’ may be used uninitialized [-Werror=maybe-uninitialized]
 1155 |   __builtin_ia32_maskstored256 ((__v8si *)__X, (__v8si)__M, (__v8si)__Y);
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc: In function ‘void fbgemm::internal::SparseDenseInt8MMAvx2(int, const std::unique_ptr<fbgemm::BCSRMatrix<> >&, const uint8_t*, int, int32_t*, uint8_t*, int, fbgemm::trRequantizationParams_t&, bool, int, int) [with bool FUSE_RELU = true; fbgemm::QuantizationGranularity Q_GRAN = fbgemm::QuantizationGranularity::OUT_CHANNEL]’:
/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc:165:19: note: ‘mask_int32_v’ was declared here
  165 |           __m256i mask_int32_v;
      |                   ^~~~~~~~~~~~
In function ‘void _mm256_maskstore_epi32(int*, __m256i, __m256i)’,
    inlined from ‘void fbgemm::internal::SparseDenseInt8MMAvx2(int, const std::unique_ptr<fbgemm::BCSRMatrix<> >&, const uint8_t*, int, int32_t*, uint8_t*, int, fbgemm::trRequantizationParams_t&, bool, int, int) [with bool FUSE_RELU = false; fbgemm::QuantizationGranularity Q_GRAN = fbgemm::QuantizationGranularity::TENSOR]’ at /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc:190:35:
/usr/lib/gcc/x86_64-pc-linux-gnu/12.1.1/include/avx2intrin.h:1155:32: error: ‘mask_int32_v’ may be used uninitialized [-Werror=maybe-uninitialized]
 1155 |   __builtin_ia32_maskstored256 ((__v8si *)__X, (__v8si)__M, (__v8si)__Y);
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc: In function ‘void fbgemm::internal::SparseDenseInt8MMAvx2(int, const std::unique_ptr<fbgemm::BCSRMatrix<> >&, const uint8_t*, int, int32_t*, uint8_t*, int, fbgemm::trRequantizationParams_t&, bool, int, int) [with bool FUSE_RELU = false; fbgemm::QuantizationGranularity Q_GRAN = fbgemm::QuantizationGranularity::TENSOR]’:
/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc:165:19: note: ‘mask_int32_v’ was declared here
  165 |           __m256i mask_int32_v;
      |                   ^~~~~~~~~~~~
In function ‘void _mm256_maskstore_epi32(int*, __m256i, __m256i)’,
    inlined from ‘void fbgemm::internal::SparseDenseInt8MMAvx2(int, const std::unique_ptr<fbgemm::BCSRMatrix<> >&, const uint8_t*, int, int32_t*, uint8_t*, int, fbgemm::trRequantizationParams_t&, bool, int, int) [with bool FUSE_RELU = false; fbgemm::QuantizationGranularity Q_GRAN = fbgemm::QuantizationGranularity::OUT_CHANNEL]’ at /home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc:190:35:
/usr/lib/gcc/x86_64-pc-linux-gnu/12.1.1/include/avx2intrin.h:1155:32: error: ‘mask_int32_v’ may be used uninitialized [-Werror=maybe-uninitialized]
 1155 |   __builtin_ia32_maskstored256 ((__v8si *)__X, (__v8si)__M, (__v8si)__Y);
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc: In function ‘void fbgemm::internal::SparseDenseInt8MMAvx2(int, const std::unique_ptr<fbgemm::BCSRMatrix<> >&, const uint8_t*, int, int32_t*, uint8_t*, int, fbgemm::trRequantizationParams_t&, bool, int, int) [with bool FUSE_RELU = false; fbgemm::QuantizationGranularity Q_GRAN = fbgemm::QuantizationGranularity::OUT_CHANNEL]’:
/home/bill/.cache/paru/clone/python-pytorch-rocm/src/pytorch-1.12.1-rocm/third_party/fbgemm/src/FbgemmSparseDenseInt8Avx2.cc:165:19: note: ‘mask_int32_v’ was declared here
  165 |           __m256i mask_int32_v;
      |                   ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
[38/2794] Building CXX object third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/FbgemmI8DepthwiseAvx2.cc.o
ninja: build stopped: subcommand failed.

```

**Comment by acxz** [comment]
Thx for the report, however, please open another issue for that.
